src/ILVerification/src/AccessVerificationHelpers.cs

```diff
@@ -216,6 +216,12 @@ private static bool CanAccessMethodSignature(this TypeDesc currentType, MethodDe
 
         private static bool CanAccessFamily(TypeDesc currentType, TypeDesc targetTypeDef, TypeDesc instanceType)
         {
+            // if instanceType is generics and inherit from targetTypeDef members of targetTypeDef are accessible
+            if (instanceType.IsGenericParameter && instanceType.CanCastTo(targetTypeDef))
```

Comment:
Do we also need to handle gracefully the case when we come here with generic parameter and `CanCastTo` returns false?

Reply:
I'll try to find a sample

Reply:
@jkotas I changed strategy, I tried to find some sample with "generic item that call a protected method but cannot cast valid base type" and the only sample I found is "Invalid" method(because this mean call a method that not exists and constraint wasn't emitted correctly) that will be catched "before" by other method check and raise method not existent validation output. I could have otherwise do some change below on buggy code like `instanceType = (instanceType.GetTypeDefinition() as MetadataType)?.ContainingType;` to void exception, but I prefer not change original alg idea. If I'm not mistaken family(protected) method can be called only by typeself or types that inherits from it, if not so should be ever "false", so I simply returned check result!
What do you think?Do you have(just every time :smile:) some "sample" that invalidate my thesis?